<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Paul Deveau" />

<meta name="date" content="2017-08-03" />

<title>Release</title>






<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Release</h1>
<h4 class="author"><em>Paul Deveau</em></h4>
<h4 class="date"><em>08/03/2017</em></h4>



<h3>
Release of 8/03/2017
</h3>
<ul>
<li>
Added normalization by contamination if “keep.all.model”
</li>
<li>
Added clause to check if normalized.center exist in evolution plot
</li>
</ul>
<h3>
Release of 25/10/2016
</h3>
<ul>
<li>
Added normalization by contamination if “keep.all.model”
</li>
<li>
Added clause to check if normalized.center exist in evolution plot
</li>
</ul>
<h3>
Release of 2/10/2016
</h3>
<ul>
<li>
Added corrective term for weight of each cluster;
</li>
<li>
Switched back default model selection to “BIC”.
</ul>
<h3>
Release of 29/9/2016
</h3>
<b> Second important update </b>
<ul>
<li>
Changed optimization procedures (see <code>One_step_clustering</code> documentation);
</li>
<li>
Added grzero: returns value that zeros the gradient (used in diploid cases);
</li>
<li>
Added tests.
</li>
</ul>
<h3>
Release of 27/9/2016
</h3>
<b> Massive update </b>
<ul>
<li>
Fixed gradient computation;
</li>
<li>
Removed alpha (to force possibility)
</li>
<li>
Switched from computation of dbinom to log(dbinom) by default;
</li>
<li>
Added testthat : plots, QuantumCat,Flash for BIC and overdiploid;
</li>
<li>
Fixed QuantumCat with single sample
</li>
</ul>
<h3>
Release of 23/9/2016
</h3>
<ul>
<li>
Added computation of gradient for “optim” and “optimx”;
</li>
<li>
Corrected priors creation from FLASH for QuantumClone;
</li>
<li>
Added jitter if number of initializations is &gt;1 and preclustering with FLASH.
</li>
</ul>
<h3>
Release of 22/9/2016
</h3>
<ul>
<li>
Fixed errors from FlashQC;
</li>
<li>
Updated validation tools.
</li>
</ul>
<h3>
Release of 20/9/2016
</h3>
<ul>
<li>
Added integration of binomial density over 2xEpsilon interval;
</li>
<li>
Added computation of Precision, Recall, F1, Rand Index in <code>Precision_Recall</code>;
</li>
<li>
Changed <code>maxit</code> to <code>Initializations</code>.
</li>
</ul>
<h3>
Release of 12/9/2016
</h3>
<ul>
<li>
Changed parameters for DEoptim controls;
</li>
<li>
Solved tree generation issue if the progeny can only come from one node.
</li>
</ul>
<h3>
Release of 12/9/2016
</h3>
<ul>
<li>
Solved fik*log(fik) when fik = 0 issue in FlashQC;
</li>
<li>
Added controls to DEoptim;
</li>
<li>
Prepared for integration of RcppDE.
</li>
</ul>
<h3>
Release of 12/9/2016
</h3>
<ul>
<li>
Changes in weight computation: normalization by sum of weights;
</li>
<li>
Added <a href="https://www.jstatsoft.org/article/view/v040i06">DEoptim </a> solver for EM;
</li>
<li>
Flash QuantumClone handles polyploid regions;
</li>
<li>
Flash QuantumClone handles BIC, AIC or numeric correction of BIC for cluster selection.
</li>
</ul>
<h3>
Release of 9/9/2016
</h3>
<h4>
MAY BE UNSTABLE
</h4>
<ul>
<li>
Added FlashQC : using hierarchical clustering and distance on the p-value of 2 variants being from the same distribution;
</li>
<li>
Updated parallel: initialization and number of clusters are run in parallel, not independently;
</li>
<li>
For rearranged genomes, after the first clustering, a second one is done after removing unlikely states;
</li>
<li>
Preclustering can now be done with FlashQC
</li>
</ul>
<h3>
Release of 31/08/2016
</h3>
<ul>
<li>
Rewrote this vignette in html
</li>
<li>
Added reproducible testing for PyClone, sciClone and k-medoid
</li>
<li>
Updated figure dpi for vignettes
</li>
</ul>
<h3>
Release of 19/08/2016:
</h3>
<ul>
<li>
Corrected NMI computation if a number of cluster is missing (for example cluster values can be 1:3, 5:7, cluster value 6 would be considered missing);
</li>
<li>
Changed the proof of concept pipeline to be easier to read and edit;
</li>
<li>
Changes in <code>Probability.to.belong.to.clone</code>: now returns filtered and unfiltered probabilities (clustering is on filtered probabilities).
</li>
</ul>
<h3>
Release of 16/08/2016:
</h3>
<ul>
<li>
Corrected hard clustering parameters;
</li>
<li>
Corrected <code>Probability.to.belong.to.clone</code>;
</li>
<li>
Gain from parallelization script.
</li>
</ul>
<h3>
Release of 12/08/2016:
</h3>
<ul>
<li>
Computed PoC for clustering quality;
</li>
<li>
Changes in <code>eval.fik</code> for an expected 5-10% gain in computational speed.
</li>
</ul>
<h3>
Release of 9/08/2016:
</h3>
<ul>
<li>
Removed 3D plots to prepare CRAN release
</li>
</ul>
<h3>
Release of 8/08/2016:
</h3>
<ul>
<li>
Removed rgl dependency (3D plots), causing QuantumClone to be unavailable on Ubuntu;
</li>
<li>
Updated pipeline tests;
</li>
<li>
Computational time test compiled;
</li>
<li>
Ongoing:
</li>
<ul>
<li>
Issue in the distance error (position of the cluster for drivers vs real position);
</li>
<li>
Compile Rmd for computational clustering quality results.
</li>
</ul>
</ul>
<h3>
Release of 4/08/2016:
</h3>
<ul>
<li>
Added function to easily compute Normalized Mutual Information from QuantumClone output - for simulated data;
</li>
<li>
Added pipelines to compare to paper pipeline.
</li>
</ul>
<h3>
Release of 29/07/2016:
</h3>
<ul>
<li>
Option to force single copy status - i.e. integrate QuantumCloneSingle in QuantumClone
</li>
<li>
Perspectives:
</li>
<ul>
<li>
tidy plot code in cluster_plot;
</li>
<li>
Reproducible test in R (QC only);
</li>
<li>
Reproducible testing in R+bash (QC+sciclone+pyClone)
</li>
</ul>
</ul>
<h3>
Release of 28/07/2016:
</h3>
<ul>
<li>
Added AIC (Akaike Information Criterion);
</li>
<li>
Added modified BIC (user input);
</li>
<li>
Added possibility to use <code>optimx</code> instead of <code>optim</code> for the maximization step;
</li>
<li>
Using <code>compiler</code> package to improve computing time;
</li>
<li>
Improved code readability;
</li>
<li>
Improved parallel computation handling;
</li>
<li>
Solved conflicting name in simulation: cellularity of clone is now refered to as cellularit;
</li>
<li>
Added option to keep all iteration models;
</li>
</ul>
<h3>
Release of 10/12/2015:
</h3>
<ul>
<li>
Changed graph output to pdf;
</li>
<li>
Fixed non-clustered graph name file
</li>
</ul>
<h3>
Release of 12/11/2015:
</h3>
<ul>
<li>
Added release tag on github;
</li>
<li>
Checked for ggplot2 scheduled release
</li>
</ul>
<h3>
Release of 2/11/2015:
</h3>
<ul>
<li>
Correction of evolution plot;
</li>
<li>
Adding content to vignettes;
</li>
<li>
Added phylogenetic tree reconstruction and representation, along with examples;
</li>
<li>
Added ImportFrom according to notes from R-devel: <a href="http://win-builder.r-project.org/" class="uri">http://win-builder.r-project.org/</a>;
</li>
</ul>
<h3>
Release of 30/10/2015:
</h3>
<h4>
Major fix
</h4>
<ul>
<li>
Post processing affected order of mutations but not of clustering. The clustering thus appeared erroneous in version 0.9.15 and 0.10.15;
</li>
</ul>
<h4>
Minor changes
</h4>
<ul>
<li>
Adding vignette release;
</li>
<li>
Changed dependency of doSNOW to doParallel;
</li>
<li>
Added data to decrease loading and check time;
</li>
<li>
Added function to plot results from output (by 2 elements);
</li>
<li>
Added function to plot cellularity of clone in each sample (for time series, with proportion of mutation in each clone);
</li>
</ul>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
